### PR TITLE
engine(dm): remove job id prefix in metadata

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: ref #5504 

### What is changed and how it works?
- remove job id prefix in metadata

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/jobmaster/dm/metadata/cluster_info_test.go [hunk 1]
```diff
@@ -29,12 +29,12 @@ func TestInfoStore(t *testing.T) {
 	clusterInfo := NewClusterInfo(ver1)
 	require.True(t, clusterInfo.Version.Equal(ver1))
 
-	clusterInfoStore := NewClusterInfoStore("cluster_info_test", mock.NewMetaMock())
+	clusterInfoStore := NewClusterInfoStore(mock.NewMetaMock())
 	key := clusterInfoStore.Key()
 	keys, err := adapter.DMInfoKeyAdapter.Decode(key)
 	require.NoError(t, err)
 	require.Len(t, keys, 1)
```
Comment (lance6716) [inline engine/jobmaster/dm/metadata/cluster_info_test.go:36]: should we return zero length?

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: engine/jobmaster/dm/metadata/cluster_info_test.go (hunk 1 above)
Comment (GMHDBJD) [reply engine/jobmaster/dm/metadata/cluster_info_test.go:36]: return [""] now, same as dm keyadater.

**Review by lance6716 — APPROVED** [review]

**Review by maxshuang — APPROVED** [review]

**Pull request merged**